File: dotnet-desktop-guide/net/winforms/input-mouse/snippets/how-to-simulate-events/csharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Runtime.InteropServices;

namespace project
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        //<ClickCheckbox>
        private void button1_Click(object sender, EventArgs e)
        {
            InvokeOnClick(checkBox1, EventArgs.Empty);
        }
        //</ClickCheckbox>

        private void SimulateButtonClick()
        {
            //<PerformClick>
            button1.PerformClick();
            //</PerformClick>
        }


        private void Form1_Load(object sender, EventArgs e)
        {

        }
    }
}

File: dotnet-desktop-guide/net/winforms/input-mouse/snippets/how-to-simulate-events/vb/Form1.vb
Imports System.Windows.Forms
Imports System.Drawing

Partial Public Class Form1

    Public Sub New()
        InitializeComponent()

    End Sub

    '<ClickCheckbox>
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        InvokeOnClick(CheckBox1, EventArgs.Empty)
    End Sub
    '</ClickCheckbox>

    Private Sub SimulateButtonClick()
        '<PerformClick>
        Button1.PerformClick()
        '</PerformClick>
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click

    End Sub

End Class
